modules/core/02-client/exported/expected_keepers.go

```diff
@@ -0,0 +1,73 @@
+package exported
+
+import (
+	"context"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	clienttypes "github.com/cosmos/ibc-go/v5/modules/core/02-client/types"
+	"github.com/cosmos/ibc-go/v5/modules/core/exported"
+)
+
+// ClientKeeper defines the expected interface of the light client used for IBC
+type ClientKeeper interface {
```

Comment:
Good point. I was thinking of the same question, given that many interfaces are put under `ibc-go/modules/core/exported`. I ended up putting it under `modules/core/02-client/exported` because `ibc-go/modules/core/exported/client.go` (which in fact defines `ClientStates`, i.e., the client of a single CZ, rather than the client keeper) was reserved. Perhaps for consistency let's move `modules/core/02-client/exported/expected_keepers.go` to `ibc-go/modules/core/exported/client_keeper.go`. 👍

Reply:
Trying it now and it looks like we cannot do that. `modules/core/02-client/types` imports `ibc-go/modules/core/exported` as a dependency. the client keeper interface imports `modules/core/02-client/types` as a dependency. If we put the client keeper interface under `ibc-go/modules/core/exported` then there will be an import cycle.

Reply:
Thanks for trying. It looks like the ibc developers never thought of abstracting the client, which makes it difficult for us to make add-on changes. Let's keep it there then.